modules/www/manifests/code_submitter.pp

```diff
@@ -23,6 +22,14 @@
     require => Vcsrepo[$root_dir],
   }
 
+  file { '/etc/sr/code-submitter-credentials.yaml':
```

Comment:
I think this needs an explicit dependency on the creation of `/etc/sr`, which may itself need pulling upwards out of the IDE-specific portion of the config.
A reasonable alternative would be to put this alongside the source of the code submitter, perhaps as `$credentials_yaml = "${root_dir}/credentials.yaml"` (and then use `$credentials_yaml` in the template).

If we go for the former approach (no variables) then I'd be tempted to also change the template to a simpler file copy (involves moving and renaming the file within this puppet repo) to clarify that it's not expecting to do any templating.

Reply:
https://github.com/srobo/server-puppet/pull/1/commits/d5942735267b88a551607760f3024d02e1b7b3a2